spec/requests/brapi/v1/phenotypes_spec.rb

```diff
@@ -0,0 +1,163 @@
+      }
+      post '/brapi/v1/phenotypes-search', {
+        germplasmDbIds: ['hzau2003_TN077_a03','hzau2004_TN038_a02'],
+        sortBy: "germplasmDbIds", 
+        sortOrder: "desc"
+      }.to_json, headers
+      expect(response.status).to eq 200
+      expect(parsed_response['result']['data'].size).to eq 2
+      expect(parsed_response['result']['data'][0]['germplasmDbId']).to eq "hzau2004_TN038_a02"
+    end
+ 
+    it 'cheking format' do        
+      headers = { 
+        "CONTENT_TYPE" => "application/json",
+        "X-BIP-Api-Key" => api_key.token 
+      }
+      post '/brapi/v1/phenotypes-search', {
+        germplasmDbIds: 'hzau2003_TN077_a03'
+      }.to_json, headers
+      expect(response.status).to eq 200
+      expect(response.header['Content-Type']).to include 'application/json'
+      results = parsed_response['result']
+      
+      result = results['data'][0]
+      
+      expect(result['observationUnitDbId']).to be_a String      # mandatory field
+      expect(result['studyDbId']).to be_a String                # mandatory field
+      expect(result['studyName']).to be_nil.or(be_a String)
+      expect(result['studyLocationDbId']).to be_a String        # mandatory field
+      expect(result['studyLocation']).to be_nil.or(be_a String)
+      expect(result['observationLevel']).to be_nil.or(be_a String)
+      expect(result['observationLevels']).to be_nil.or(be_a String)
+      expect(result['plotNumber']).to be_nil.or(be_a String)
+      expect(result['plantNumber']).to be_nil.or(be_a String)
+      expect(result['blockNumber']).to be_nil.or(be_a String)
+      expect(result['replicate']).to be_nil.or(be_a String)
+      expect(result['programName']).to be_nil.or(be_a String)
+      expect(result['germplasmDbId']).to be_a String            # mandatory field
+
+      expect(result['X']).to be_nil.or(be_a String)
+      expect(result['Y']).to be_nil.or(be_a String)
+      expect(result['treatments']).to be_nil.or(be_a Array)
+      expect(result['observationUnitXref']).to be_nil.or(be_a Array)
+      if (result['observationUnitXref'].is_a? Array)
+        observationsXref = result['observationUnitXref']
+        
+        if (observationsXref.size > 0)
+          observationXref = observationsXref[0]
+          
+          expect(observationXref).to be_nil.or(be_a Hash)
+          expect(observationXref['source']).to be_a String     # mandatory field
+          expect(observationXref['id']).to be_a String         # mandatory field
+
+          observationXref.keys.each do |key|
+            if (['source','id'].include? key)
+              expect(observationXref[key]).to be_nil.or(be_a String)
+            else
+              raise_error('Unexpected property name: observationXref['+key+']')
+            end
+          end  
+        end
+      end
+      
+      expect(result['observations']).to be_a Array             # mandatory field
+      if (result['observations'].is_a? Array)
```

Comment:
If it is not an array then the expectation in line 135 will fail and nothing more will be executed.